### PR TITLE
Allow "rush version --ensure-version-policy" to update policy version directly

### PR DESCRIPTION
**Pull request by qz2017**
It can be useful when the policy version needs to be reset to a different value.

### CODE REVIEW

File: common/reviews/api/rush-lib.api.ts [hunk 1]
```diff
@@ -149,6 +149,7 @@ class LockStepVersionPolicy extends VersionPolicy {
   ensure(project: IPackageJson): IPackageJson | undefined;
   readonly mainProject: string | undefined;
   readonly nextBump: BumpType;
+  update(newVersion: semver.SemVer): boolean;
```
Comment (pgonzal) [inline common/reviews/api/rush-lib.api.ts]: >semver.SemVer [](start = 21, length = 13)

Can we eliminate this dependency on the "semver" API contract?  That seems like a complex and unnecessary coupling

File: apps/rush-lib/src/data/VersionPolicyConfiguration.ts [hunk 2]
```diff
@@ -115,12 +116,24 @@ export class VersionPolicyConfiguration {
...
+   * Updates the version directly for the specified version policy
+   * @param versionPolicyName - version policy name
+   * @param newVersion - new version
+   */
+  public update(versionPolicyName: string,
+    newVersion: semver.SemVer
+  ): void {
+    const policy: VersionPolicy | undefined = this.versionPolicies.get(versionPolicyName);
+    if (!policy || !policy.isLockstepped) {
+      throw new Error(`Lockstep Version policy with name ${versionPolicyName} cannot be found`);
```
Comment (pgonzal) [inline apps/rush-lib/src/data/VersionPolicyConfiguration.ts]: >${versionPolicyName} [](start = 57, length = 20)

add double-quotes

File: apps/rush-lib/src/cli/actions/VersionAction.ts [hunk 3]
```diff
@@ -48,9 +50,14 @@ export default class VersionAction extends BaseRushAction {
     });
     this._ensureVersionPolicy = this.defineFlagParameter({
       parameterLongName: '--ensure-version-policy',
-      parameterShortName: '-e',
```
Comment (pgonzal) [inline apps/rush-lib/src/cli/actions/VersionAction.ts]: >parameterShortName: '-e', [](start = 6, length = 25)

Do we need any of these short names?  I already updated all our CI definitions to always use long names.

Comment (qz2017) [reply apps/rush-lib/src/cli/actions/VersionAction.ts]: That line is deleted.

---
In reply to: [183606000](https://github.com/Microsoft/web-build-tools/pull/619#discussion_r183606000) [](ancestors = 183606000)

File: common/changes/@microsoft/rush/qz2017-update_version_2018-04-23-21-25.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Allow \\\"rush version --ensure-version-policy\\\" to update policy version directly. ",
```
Comment (pgonzal) [inline common/changes/@microsoft/rush/qz2017-update_version_2018-04-23-21-25.json]: >\\ [](start = 24, length = 2)

Extra backslash

**Pull request merged**